worker/src/bin/worker.rs

```diff
@@ -110,24 +103,26 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let node_name = std::env::var("NODE_NAME")
         .unwrap_or_else(|_| gethostname::gethostname().to_str().unwrap().to_owned());
 
+    let port = args.port.unwrap_or(0);
+
     let my_url = if let Some(my_url) = args.my_url.as_ref() {
         Some(my_url.clone())
     } else if let Some(external_address) = args.external_address {
         Some(
-            format!("http://{}:{}", external_address, args.site_port)
+            format!("http://{}:{}", external_address, port)
```

Comment:
This isn't obtained from the socket, so will be 0

Reply:
Gotcha - Ive re-worked the code to now (hopefully) support it this time around :)